solidity/src/interfaces/CrossVMBridgeERC721Fulfillment.sol

```diff
@@ -0,0 +1,111 @@
+ * integrate with the Flow VM bridge connecting Cadence & EVM implementations so that the canonical
+ * VM bridge may move the Cadence NFT into EVM in a mint/escrow pattern.
+ */
+abstract contract CrossVMBridgeERC721Fulfillment is ICrossVMBridgeERC721Fulfillment, CrossVMBridgeCallable, ERC721 {
+
+    // Encountered when attempting to fulfill a token that has been previously minted and is not
+    // escrowed in EVM under the VM bridge
+    error FulfillmentFailedTokenNotEscrowed(uint256 id, address escrowAddress);
+
+    // Emitted when an NFT is moved from Cadence into EVM
+    event FulfilledToEVM(address indexed recipient, uint256 indexed tokenId);
+
+    /**
+     * Initializes the bridge EVM address such that only the bridge COA can call privileged methods
+     */
+    constructor(address _vmBridgeAddress) CrossVMBridgeCallable(_vmBridgeAddress) {}
+
+    /**
+     * @dev Fulfills the bridge request, minting (if non-existent) or transferring (if escrowed) the
+     * token with the given ID to the provided address. For dynamic metadata handling between
+     * Cadence & EVM, implementations should override and assign metadata as encoded from Cadence
+     * side. If overriding, be sure to preserve the mint/escrow pattern as shown in the default
+     * implementation.
+     * 
+     * @param _to address of the token recipient
+     * @param _id the id of the token being moved into EVM from Cadence
+     * @param _data any encoded metadata passed by the corresponding Cadence NFT at the time of
+     *      bridging into EVM
+     */
+    function fulfillToEVM(address _to, uint256 _id, bytes memory _data) external onlyVMBridge {
+        _beforeFulfillment(_to, _id, _data); // hook allowing implementation to perform pre-fulfillment validation
+        if (_ownerOf(_id) == address(0)) {
+            _mint(_to, _id); // Doesn't exist, mint the token
+        } else {
+            // Should be escrowed under vm bridge - transfer from escrow to recipient
+            _requireEscrowed(_id);
+            safeTransferFrom(vmBridgeAddress(), _to, _id);
+        }
+        _afterFulfillment(_to, _id, _data); // hook allowing implementation to perform post-fulfillment processing
+        emit FulfilledToEVM(_to, _id);
+    }
+
+    /**
+     * @dev Returns whether the token is currently escrowed under custody of the designated VM bridge
+     * 
+     * @param _id the ID of the token in question
+     */
+    function isEscrowed(uint256 _id) public view returns (bool) {
+        return _ownerOf(_id) == vmBridgeAddress();
+    }
+
+    /**
+     * @dev Allows a caller to determine the contract conforms to the `ICrossVMFulfillment` interface
+     */
+    function supportsInterface(bytes4 interfaceId) public view virtual override(CrossVMBridgeCallable, ERC721, IERC165) returns (bool) {
+        return interfaceId == type(CrossVMBridgeERC721Fulfillment).interfaceId
```

Comment:
could `type(CrossVMBridgeERC721Fulfillment).interfaceId` be removed since there is already `type(ICrossVMBridgeERC721Fulfillment).interfaceId`?

Reply:
Yes, I suppose that's sufficient. I don't think the `onlyVMBridge` modifier is included in the function signature anyway which is the most important aspect of the contract.

Reply:
Updated in [274a2c3](https://github.com/onflow/flow-evm-bridge/pull/168/commits/274a2c3bb9fe428edb58991ae122d812d39570c0)